docs/Lexicon.md

```diff
@@ -193,6 +212,17 @@ compared directly.
 
 ["if and only if"](https://en.wikipedia.org/wiki/If_and_only_if). This term comes from mathematics.
 
+## implicit module build
+
+A module build where the compiler is free to transparently build dependent
+modules, and access modules in different caches as necessary. For example,
+if a textual Swift module (.swiftinterface file) for a dependency does not
+have a corresponding binary Swift module (.swiftmodule file), the compiler
+may transparently build a binary Swift module from the textual one as a cache
+for future compiler jobs, without involving any external build system that
+invoked the compiler.  See also:
+[explicit module build](#explicit-module-build).
```

Comment:
Please also mention the shortcomings of implicit module builds: because build systems have no way to control when and where these modules are built, sharing modules among different nodes for distributed build systems can be difficult.